### PR TITLE
Add redirect field to 1.1 spec

### LINKED ISSUE #162: [feature request] Support for redirect urls on txn success
Wallet providers who support Solana Pay can allow user redirecting on successful txns, just like deep link implementations. This UX will make it familiar with GPay, Apple Pay, etc, and can provide merchants or payment providers a room to support new features. 

**Motivation**:  - Recently at Delhi HH it was my first experience seeing people actually mint NFTs IRL by scanning a Solana pay QR Code. While the experience was seamless, post minting, new users who created their first ever wallet had no idea what to further do. Wallet providers take some time to notify users that they got an NFT and further loading of this NFT image takes a lot of time, meanwhile the end user remains clueless. 
- Similarly, there can be a case where a user paid to buy something with Solana pay on web apps (not taking pos reference here) but is confused if their user was actually marked paid or not. 
- Businesses accepting Solana pay will expect their user to come back to their site and keep surfing more post a successful order, instead of remaining on the wallet provider apps, without a specific action on what to do next
- NFT and Payment Providers would like to show users with actual NFT minted and/or the invoice of payment made respectively as soon as the txn is confirmed, instead of having them wait and figure things out themselves 

**Implementation**:  - There's little to no change on the protocol level for this feature, and most work will be on the wallet provider side.  As a user, we can pass a  `redirect_url` param in the `POST` response along with other parameters like `message`, which can give us an extra space to make Redirect URLs more targeted, e.g. passing mint address for newly minted NFT with an explorer link by which user can see metadata for their minted NFTs instantly. 

Excited to hear everyone's feedback and views on this addition. Thanks to @cogoo & @mcintyre94 for listening to this idea initially, and providing their input!

### PR DESCRIPTION
**Pull request by mcintyre94**
This PR adds `redirect` as an optional field for the transfer request, transaction request and message signing specs.

Closes #162

### CODE REVIEW

**Review by jordansexton — COMMENTED** [review]
Made a short review, will make another pass!

**Review by jordansexton — APPROVED** [review]
Looks good!

File: SPEC1.1.md [hunk 1]
```diff
@@ -95,6 +103,11 @@ solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=0.01&spl-token=EPjFWdd
 solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN&label=Michael
 ```
 
+##### URL describing a transfer request for 1 SOL with a success redirect
+```
+solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=1&label=Michael&message=Thanks%20for%20all%20the%20fish&memo=OrderId12345&redirect=https://example.com
```
Comment (jordansexton) [inline SPEC1.1.md:108]: nit: `https://example.com` needs to be URL-encoded

File: SPEC1.1.md [hunk 2]
```diff
@@ -187,10 +200,22 @@ The `<message>` value must be a UTF-8 string that describes the nature of the tr
 
 For example, this might be the name of an item being purchased, a discount applied to the purchase, or a thank you note. The wallet should display the value to the user.
 
+The application may also include an optional `redirect` field in the response body:
+
+```json
+{"redirect":"<redirect>","transaction":"<transaction>"}
```
Comment (jordansexton) [inline SPEC1.1.md:206]: Maybe we want to include these in the example HTTP requests/responses?

File: message-signing-spec.md [hunk 3]
```diff
@@ -90,6 +90,15 @@ The `<message>` value must be a UTF-8 encoded string that describes the nature o
...
 For example, this might be the name of the application with which the user is interacting or context about how the sign-message request is being used. The wallet must display at least the first 80 characters of the message field to the user if it is included in the response.
 
+The application may also include an optional `redirect` field in the response body:
+```json
+{"redirect":"<redirect>", "message":"<message>","data":"<data>","state":"<state>"}
+```
+
+The `redirect` field must be a [URL-encoded](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) absolute HTTPS or solana URL.
+
+The wallet must [URL-decode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent) the value. If it is a HTTPS URL then the wallet should display the decoded value to the user.
```
[the remark this answers is not in the document]
Comment (mcintyre94) [reply message-signing-spec.md]: Updated here and also the transaction request post response

**Pull request merged**